### PR TITLE
20 dictionary contains key tomlin

### PR DESCRIPTION
**Pull request by TomlinMichael**
Adding dictionary contains key activity to main (please review)

### CODE REVIEW

File: UiPathTeam.Extensions.Tests/DictionaryUnitTests.cs [hunk 1]
```diff
@@ -99,6 +99,77 @@ public void TestDictionaryContainsValue()
 
             Assert.IsTrue(boolResult);
         }
+		[TestMethod]
+        public void TestDictionaryContainsKey_KeyExists()
```
Comment (k2zinger) [inline UiPathTeam.Extensions.Tests/DictionaryUnitTests.cs:103]: this unit test is incorrect. you added "key1" to the dictionary and are testing to see if "Key1" exists, which is false.

File: UiPathTeam.Extensions/UiPathTeam.Extensions.Activities/Activities/DictionaryContainsKey.cs [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+            base.CacheMetadata(metadata);
+        }
+
+        protected override async Task<Action<AsyncCodeActivityContext>> ExecuteAsync(AsyncCodeActivityContext context, CancellationToken cancellationToken)
+        {
+            // Inputs
+            var in_dictionary = In_dictionary.Get(context);
+            var in_key = In_key.Get(context);
+            var out_result = Out_result.Get(context);
```
Comment (k2zinger) [inline UiPathTeam.Extensions/UiPathTeam.Extensions.Activities/Activities/DictionaryContainsKey.cs:71]: Out_result.Get(context) is always null; this assignment is unnecessary

File: UiPathTeam.Extensions/UiPathTeam.Extensions.Activities/Activities/DictionaryContainsKey.cs [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+            var out_value = Out_value.Get(context);
```
Comment (k2zinger) [inline UiPathTeam.Extensions/UiPathTeam.Extensions.Activities/Activities/DictionaryContainsKey.cs:72]: Out_value.Get(context) is always null; this assignment is unnecessary

File: UiPathTeam.Extensions/UiPathTeam.Extensions.Activities/Properties/Resources.resx [hunk 4]
```diff
@@ -273,4 +273,96 @@
...
+  </data>
+  <data name="DictionaryContainsKey_Out_value_DisplayName" xml:space="preserve">
+    <value>out_value</value>
+    <comment>property name</comment>
+  </data>
+  <data name="DictionaryContainsKey_Out_value_Description" xml:space="preserve">
+    <value>Dictionary value if exists</value>
+    <comment>property description</comment>
+  </data>
+  <data name="Output_Category" xml:space="preserve">
```
Comment (k2zinger) [inline UiPathTeam.Extensions/UiPathTeam.Extensions.Activities/Properties/Resources.resx:364]: duplicate, remove

File: UiPathTeam.Extensions/UiPathTeam.Extensions.Activities/Properties/Resources.resx [hunk 5]
```diff
@@ -273,4 +273,96 @@
...
+  </data>
+  <data name="DictionaryContainsKey_Out_result_DisplayName" xml:space="preserve">
+    <value>out_result</value>
+    <comment>property name</comment>
+  </data>
+  <data name="DictionaryContainsKey_Out_result_Description" xml:space="preserve">
+    <value>Bool result of existing key</value>
+    <comment>property description</comment>
+  </data>
+  <data name="Output_Category" xml:space="preserve">
```
Comment (k2zinger) [inline UiPathTeam.Extensions/UiPathTeam.Extensions.Activities/Properties/Resources.resx:352]: duplicate, remove

File: UiPathTeam.Extensions/UiPathTeam.Extensions.Activities/Properties/Resources.resx [hunk 6]
```diff
@@ -273,4 +273,96 @@
...
+  </data>
+  <data name="DictionaryContainsKey_In_key_DisplayName" xml:space="preserve">
+    <value>in_key</value>
+    <comment>property name</comment>
+  </data>
+  <data name="DictionaryContainsKey_In_key_Description" xml:space="preserve">
+    <value>Key to be evaulated</value>
+    <comment>property description</comment>
+  </data>
+  <data name="Input_Category" xml:space="preserve">
```
Comment (k2zinger) [inline UiPathTeam.Extensions/UiPathTeam.Extensions.Activities/Properties/Resources.resx:340]: duplicate, remove

**Review by k2zinger — CHANGES_REQUESTED** [review]
Can you review these items and update the pull request please?

File: UiPathTeam.Extensions/UiPathTeam.Extensions.Activities/Activities/DictionaryContainsKey.cs (hunk 2 above)
Comment (TomlinMichael) [reply UiPathTeam.Extensions/UiPathTeam.Extensions.Activities/Activities/DictionaryContainsKey.cs:71]: My code does not compile without this assignments. When I remove, i get errors below.

Severity	Code	Description	Project	File	Line	Suppression State
Error	CS0103	The name 'out_value' does not exist in the current context	UiPathTeam.Extensions.Activities (net461), UiPathTeam.Extensions.Activities (net6.0)	C:\Users\michael.tomlin\source\repos\UiPathTAMTeam.Extensions.Activities\UiPathTeam.Extensions\UiPathTeam.Extensions.Activities\Activities\DictionaryContainsKey.cs	78	Active
Error	CS0103	The name 'out_value' does not exist in the current context	UiPathTeam.Extensions.Activities (net461), UiPathTeam.Extensions.Activities (net6.0)	C:\Users\michael.tomlin\source\repos\UiPathTAMTeam.Extensions.Activities\UiPathTeam.Extensions\UiPathTeam.Extensions.Activities\Activities\DictionaryContainsKey.cs	88	Active
Error	CS0103	The name 'out_result' does not exist in the current context	UiPathTeam.Extensions.Activities (net461), UiPathTeam.Extensions.Activities (net6.0)	C:\Users\michael.tomlin\source\repos\UiPathTAMTeam.Extensions.Activities\UiPathTeam.Extensions\UiPathTeam.Extensions.Activities\Activities\DictionaryContainsKey.cs	77	Active
Error	CS0103	The name 'out_result' does not exist in the current context	UiPathTeam.Extensions.Activities (net461), UiPathTeam.Extensions.Activities (net6.0)	C:\Users\michael.tomlin\source\repos\UiPathTAMTeam.Extensions.Activities\UiPathTeam.Extensions\UiPathTeam.Extensions.Activities\Activities\DictionaryContainsKey.cs	82	Active
Error	CS0103	The name 'out_result' does not exist in the current context	UiPathTeam.Extensions.Activities (net461), UiPathTeam.Extensions.Activities (net6.0)	C:\Users\michael.tomlin\source\repos\UiPathTAMTeam.Extensions.Activities\UiPathTeam.Extensions\UiPathTeam.Extensions.Activities\Activities\DictionaryContainsKey.cs	87	Active
Error	CS0006	Metadata file 'C:\Users\michael.tomlin\source\repos\UiPathTAMTeam.Extensions.Activities\UiPathTeam.Extensions\UiPathTeam.Extensions.Activities\obj\Debug\net6.0\ref\UiPathTeam.Extensions.Activities.dll' could not be found	UiPathTeam.Extensions.Tests (net461)	C:\Users\michael.tomlin\source\repos\UiPathTAMTeam.Extensions.Activities\UiPathTeam.Extensions.Tests\CSC	1	Active
Error	CS0006	Metadata file 'C:\Users\michael.tomlin\source\repos\UiPathTAMTeam.Extensions.Activities\UiPathTeam.Extensions\UiPathTeam.Extensions.Activities\bin\Debug\net461\UiPathTeam.Extensions.Activities.dll' could not be found	UiPathTeam.Extensions.Activities.Design (net461)	C:\Users\michael.tomlin\source\repos\UiPathTAMTeam.Extensions.Activities\UiPathTeam.Extensions\UiPathTeam.Extensions.Activities.Design\CSC	1	Active
Error	CS0006	Metadata file 'C:\Users\michael.tomlin\source\repos\UiPathTAMTeam.Extensions.Activities\UiPathTeam.Extensions\UiPathTeam.Extensions.Activities\bin\Debug\net461\UiPathTeam.Extensions.Activities.dll' could not be found	UiPathTeam.Extensions.Tests (net461)	C:\Users\michael.tomlin\source\repos\UiPathTAMTeam.Extensions.Activities\UiPathTeam.Extensions.Tests\CSC	1	Active
Error		Could not find a part of the path 'C:\Users\michael.tomlin\source\repos\UiPathTAMTeam.Extensions.Activities\UiPathTeam.Extensions\UiPathTeam.Extensions.Activities.Design\obj\Debug\net6.0-windows7.0\UiPathTeam.Extensions.Activities.Design_0l5m1de5_wpftmp.GeneratedMSBuildEditorConfig.editorconfig'.	UiPathTeam.Extensions.Activities.Design	C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Current\Bin\Roslyn\Microsoft.Managed.Core.targets	191

Comment (k2zinger) [reply UiPathTeam.Extensions/UiPathTeam.Extensions.Activities/Activities/DictionaryContainsKey.cs:71]: you still need to define the variables, but they should not be assigned to values that are null
var out_result;

File: UiPathTeam.Extensions.Tests/DictionaryUnitTests.cs (hunk 1 above)
Comment (TomlinMichael) [reply UiPathTeam.Extensions.Tests/DictionaryUnitTests.cs:103]: Test dictionary contains value is Cathleen's function, mine is Test Dictionary Contains Key

File: UiPathTeam.Extensions/UiPathTeam.Extensions.Activities/Activities/DictionaryContainsKey.cs (hunk 2 above)
Comment (TomlinMichael) [reply UiPathTeam.Extensions/UiPathTeam.Extensions.Activities/Activities/DictionaryContainsKey.cs:71]: Gotcha, thanks. I believe the other activities in the project also assign a value in the same I did mistakenly. Fixed.

File: UiPathTeam.Extensions.Tests/DictionaryUnitTests.cs (hunk 1 above)
Comment (k2zinger) [reply UiPathTeam.Extensions.Tests/DictionaryUnitTests.cs:103]: this is for the unit test TestDictionaryContainsKey_KeyExists

**Review by k2zinger — APPROVED** [review]

**Pull request merged**